mkosi/__init__.py

```diff
@@ -4472,14 +4472,14 @@ def run_serve(args: Args, config: Config) -> None:
             http.send_signal(signal.SIGINT)
 
 
-def generate_key_cert_pair(args: Args) -> None:
+def generate_key_cert_pair(args: Args, configdir: Path) -> None:
     """Generate a private key and accompanying X509 certificate using openssl"""
 
     keylength = 2048
     expiration_date = datetime.date.today() + datetime.timedelta(int(args.genkey_valid_days))
 
     for f in ("mkosi.key", "mkosi.crt"):
-        if Path(f).exists() and not args.force:
+        if Path(configdir / f).exists() and not args.force:
```

Comment:
since `configdir` is by construction already a `Path`.
```suggestion
        if (configdir / f).exists() and not args.force:
```